script.js

```diff
@@ -90,11 +51,24 @@ restaurantsApp.getReviews=function(result){
         })
     }}
 
-restaurantsApp.displayRestaurantDetails = function(result) {
-  
-    $('.ListOne').empty()
+// Use the GeoLocation-DB API to get the user's city
+restaurantsApp.getCity = async function() {
+    // wait for the result to come back (promise)
+    let location = await $.ajax({
+        url: "https://geolocation-db.com/jsonp",
+        jsonpCallback: "callback",
+        dataType: "jsonp"
+    });
+
+    return location.city;
+}
+
+restaurantsApp.displayRestaurantDetails = function(result, addedResults) {
+    
+    if(addedResults==0){
```

Comment:
since there is a lot of duplicate code in both branches of this if statement, let's try to remove some of it. it looks like you're checking if `addedResults==0` so you can call `.empty()` on the restaurantList.

Can we remove the if-else statement and just have:

```
if(addedResults==0) {
$('.restaurantList').empty()
}
```

Then after we can have the for loop: 

```
for (let i = addedResults; i < addedResults + 6; i++) {
        let businessID = result.businesses[i].id
        let businessName = result.businesses[i].name
        let businessImage = result.businesses[i].image_url
        let businessRating = result.businesses[i].rating
        let businessPrice = result.businesses[i].price
        let businessAddress = result.businesses[i].location.display_address

        let html = `<div>
                <img src="${businessImage}">
                <h2>${businessName}</h2>
                <span>${businessRating}</span>
                <span><i class="fas fa-dollar-sign"></i><i class="fas fa-dollar-sign"></i></span> 
                <h3>${businessAddress}</h3>
                </div>`
        //Displays each result to the page
        $('.restaurantList').append(html)
    }}
```
this would avoid having the same loop twice

Reply:
the if statement is actually checking to see if we are displaying results on the page for the first time or else we are adding more results after the user reaches the bottom of the page. (using the resultsAdded counter) if I use .empty() the new results would not append to the one displayed before.

Reply:
Yes this makes sense. But the loops are doing exactly the same thing aren't they? So you could have a single if-statement at the top that checks if we are doing it for the first time or not and calls `.empty()`. 

Then after that, you would just have one for-loop.